source/Playnite/Scripting/PowerShell/PowerShell.cs

```diff
@@ -28,19 +32,13 @@ public static bool IsInstalled
 
         public PowerShellRuntime(string runspaceName = "PowerShell")
         {
-            runspace = RunspaceFactory.CreateRunspace();
+            initialSessionState = InitialSessionState.CreateDefault();
+            initialSessionState.ExecutionPolicy = ExecutionPolicy.Unrestricted;
+            initialSessionState.ThreadOptions = PSThreadOptions.UseCurrentThread;

```

Comment:
I tried to figure out how to get `System.Management.Automation.PowerShell` to work with an MTA apartment state, but couldn't get it to work well. As I mentioned in a previous comment, I get this error if I set it:
>When the runspace is set to use the current thread, the apartment state in the invocation settings must match that of the current thread.

The following allows MTA to work, but breaks usage of GUI from extensions:
```
initialSessionState.ApartmentState = System.Threading.ApartmentState.MTA;
initialSessionState.ThreadOptions = PSThreadOptions.ReuseThread;
```
LibraryExporter just hangs after clicking its menu item. The file dialog never appears.

I don't know STA vs MTA differences well enough. What are the implications of using STA? Will that break existing extensions?

It might be possible to manually create the runspace like the old code did to get the old behavior back.

Reply:
I tried manually creating the runspace like this, but it doesn't work either. I get the same error.
```csharp
initialSessionState = InitialSessionState.CreateDefault();
initialSessionState.ExecutionPolicy = ExecutionPolicy.Unrestricted;

runspace = RunspaceFactory.CreateRunspace(initialSessionState);
runspace.Name = runspaceName; runspace.Name = runspaceName;
runspace.ApartmentState = System.Threading.ApartmentState.MTA;
runspace.ThreadOptions = PSThreadOptions.UseCurrentThread;
runspace.Open();

powershell = System.Management.Automation.PowerShell.Create();
powershell.Runspace = runspace;
```

Reply:
We have to use STA because of GUI interaction, there's no workaround as far as I know.

Reply:
I think the only solution would be to create separate thread for PS extensions and then re-implementing every GUI call to be thread safe. But I don't think that's worth the trouble, because Playnite always has to wait for script to finish anyways, so the execution will always act as synchronous not matter what. We should also keep it synchronous to have the same execution behavior consistent with plugins.

Reply:
Okay, then we're good. It's already STA. Somehow the old PowerShell implementation was MTA and had working GUI though. Perhaps it's the new method of invoking PowerShell I'm using, but it now gives me errors with MTA+UseCurrentThread. As long as STA isn't a breaking change, I'm good with this.

Reply:
Yeah, it's because you now execute methods differently. I have sneaking suspicion that we might need to change this once we switch to PS Core (because we will ship Playnite with built-in PS runtime), but I haven't tested it yet.

Now that I'm thinking about it, this could be an issue for game scripts, because those are still being executed the old way, right? But I don't know how likely is that someone will be using GUI code there though.